db.php

```diff
@@ -34,35 +34,34 @@ public function query($query)
     {
         $this->closeOpenQuery();
 
-        if ($this->query = $this->connection->prepare($query)) {
-            if (func_num_args() > 1) {
-                $x = func_get_args();
-                $args = array_slice($x, 1);
-                $types = '';
-                $args_ref = array();
-                foreach ($args as $k => &$arg) {
-                    if (is_array($args[$k])) {
-                        foreach ($args[$k] as $j => &$a) {
-                            $types .= $this->_gettype($args[$k][$j]);
-                            $args_ref[] = &$a;
-                        }
-                    } else {
-                        $types .= $this->_gettype($args[$k]);
-                        $args_ref[] = &$arg;
+        if (!$this->query = $this->connection->prepare($query)) {

```

Comment:
The problem with this is that if `$this->show_errors` is `FALSE` it will not exit and continue to process the function arguments.